src/Uno.UI/UI/Xaml/Controls/TextBox/OverlayTextBoxView/OverlayTextBoxViewExtension.skia.cs

```diff
@@ -0,0 +1,290 @@
+﻿#nullable enable
+
+using System;
+using System.Linq;
+using Uno.Disposables;
+using Uno.UI.Extensions;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Size = Windows.Foundation.Size;
+using Point = Windows.Foundation.Point;
+
+namespace Uno.UI.Xaml.Controls.Extensions;
+
+internal abstract class OverlayTextBoxViewExtension : IOverlayTextBoxViewExtension
+{
+	private readonly TextBoxView _owner;
+	private readonly Func<TextBox, ITextBoxView> _textBoxViewFactory;
+	private readonly SerialDisposable _textChangedDisposable = new SerialDisposable();
+
+	private ContentControl? _contentElement;
+	private ITextBoxView? _textBoxView;
+	private bool _processingTextChanged;
+	private Point _lastPosition = new(-1, -1);
+	private Size _lastSize = new(-1, -1);
+
+	private int? _selectionStartCache;
+	private int? _selectionLengthCache;
+
+	public OverlayTextBoxViewExtension(TextBoxView owner, Func<TextBox, ITextBoxView> textBoxViewFactory)
+	{
+		_owner = owner ?? throw new ArgumentNullException(nameof(owner));
+		_textBoxViewFactory = textBoxViewFactory ?? throw new ArgumentNullException(nameof(textBoxViewFactory));
+	}
+
+	public abstract bool IsOverlayLayerInitialized(XamlRoot xamlRoot);
+
+	public void StartEntry()
+	{
+		if (_owner.TextBox is not { } textBox ||
+			_owner.TextBox.XamlRoot is not { } xamlRoot)
```

Comment:
Minor stylistic suggestion.
Feel free to ignore if you find it less readable.

```suggestion
		if (_owner.TextBox is not { XamlRoot: { } xamlRool } textBox)
```